### ISSUE TITLE
Generate builders

### ISSUE THREAD

**Issue by antoyo** [gtk-rs/gir#286]
Hello.
In GObject, it is possible to create an object with multiple construct properties with `g_object_new`.
I wonder if it would be nice to generate builders (using the builder pattern) for objects using their construct properties.
[Here](https://github.com/antoyo/webkit2gtk-rs/blob/master/src/web_view.rs#L55) is a constructor that I wrote because it was impossible to use it correctly if I set these properties after creating the object.

I believe we could use the builder pattern so that we could construct this particular object as such:

``` rust
let webview: WebView = WebViewBuilder::new()
    .context(context)
    .user_content_manager(content_manager)
    .build();
```

What do you think about this idea?
Do you have another idea?
Thanks.

**Comment by gkoz** [comment]
It's a good idea but probably secondary to actually generating a struct corresponding to constructor properties.

**Comment by antoyo** [comment]
Fixed by #757.
